protocol/consistencychecks.go

```diff
@@ -245,7 +245,11 @@ func (cc *ConsistencyChecks) updateTBs(requestType int, msg *Response,
 	case RegistrationType:
 		df := msg.DirectoryResponse.(*DirectoryProof)
 		if df.AP.ProofType() == m.ProofOfAbsence {
-			if err := cc.verifyReturnedPromise(df, key); err != nil {
+			tbKey := key
+			if msg.Error == ReqNameExisted {
+				tbKey = df.TB.Value
+			}
```

Comment:
> Should KeyChange be used here instead?

Ideally, but we don't have that yet.

> In this case, how does client distinguish its own registration with other client's registration?

I'm going to assume you actually do mean "client" here (ie. the "user" is trying to reregister for some legitimate reason.)  I imagine, in both cases, the client would notify the user that a different binding already exists for that account, and we'd let the user disambiguate the situation.  If they understood that they're trying to initiate a key change, they'd do that, or otherwise report that something is amiss.

In other words, I'm not sure the client itself can distinguish which situation it's in.  I think this is getting in the realm of UI, and that's something we've hardly tackled to this point.

The case where the registration succeeds but we get an `ErrBindingsDiffer` is a little more clearcut though, but I'm not sure why any deceitful server would bother doing this.

> Also, if it gets ReqNameExisted and ErrBindingsDiffer because of other user's registration, will calling GetKey() and re-verifying with HandleResponse(new_key) make sense?

Can you clarify what you mean by "other user's registration"?